spine_items/tool/widgets/options_widgets.py

```diff
@@ -247,11 +248,11 @@ def _create_sysimage(self, _checked=False):
         spec_dict["definition_file_path"] = spec.definition_file_path
         engine_data["specifications"].setdefault(self._tool.item_type(), list()).append(spec_dict)
         self.sysimage_worker.set_engine_data(engine_data)
-        self.sysimage_worker.finished.connect(lambda tool=self._tool: self._do_create_sysimage(tool))
+        self.sysimage_worker.finished.connect(lambda: self._do_create_sysimage(self._tool))
```

Comment:
I think it is better to keep storing the current self._tool in the lambda function because self._tool can change arbitrarily and we need to make sure we pass the right one to self._do_create_sysimage()
-or I am missing something?

Reply:
Nice catch! Looks like I missed the fact that `self._tool` may change.